app/screens/global_threads/threads_list/thread/thread.tsx

```diff
@@ -0,0 +1,265 @@
+            borderBottomWidth: 1,
+        },
+        badgeContainer: {
+            marginTop: 3,
+            width: 32,
+        },
+        postContainer: {
+            flex: 1,
+        },
+        header: {
+            alignItems: 'center',
+            flex: 1,
+            flexDirection: 'row',
+            marginBottom: 9,
+        },
+        headerInfoContainer: {
+            alignItems: 'center',
+            flex: 1,
+            flexDirection: 'row',
+            marginRight: 12,
+            overflow: 'hidden',
+        },
+        threadDeleted: {
+            color: changeOpacity(theme.centerChannelColor, 0.72),
+            fontStyle: 'italic',
+        },
+        threadStarter: {
+            color: theme.centerChannelColor,
+            fontSize: 15,
+            fontWeight: '600',
+            lineHeight: 22,
+            paddingRight: 8,
+        },
+        channelNameContainer: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+            borderRadius: 4,
+            maxWidth: '50%',
+        },
+        channelName: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 25, 'SemiBold'),
+            letterSpacing: 0.1,
+            textTransform: 'uppercase',
+            marginHorizontal: 12,
+            marginVertical: 2,
+        },
+        date: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 25, 'Light'),
+        },
+        message: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 200),
+        },
+        unreadDot: {
+            width: 8,
+            height: 8,
+            borderRadius: 4,
+            backgroundColor: theme.sidebarTextActiveBorder,
+            alignSelf: 'center',
+            marginTop: 5,
+        },
+        mentionBadge: {
+            width: 18,
+            height: 18,
+            borderRadius: 9,
+            backgroundColor: theme.buttonBg,
+            alignSelf: 'center',
+        },
+        mentionBadgeText: {
+            ...typography('Body', 25, 'SemiBold'),
+            alignSelf: 'center',
+            color: theme.buttonColor,
+        },
+    };
+});
+
+const Thread = ({author, channel, post, teammateNameDisplay, testID, thread}: Props) => {
+    const intl = useIntl();
+    const isTablet = useIsTablet();
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+
+    const handleMarkdownEmoji = useCallback(({emojiName, literal}: MarkdownEmojiRenderer) => {
+        return (
+            <Emoji
+                emojiName={emojiName}
+                literal={literal}
+                testID='markdown_emoji'
+                textStyle={styles.message}
+            />
+        );
+    }, [styles]);
+
+    const handleMarkdownSoftBreak = useCallback(() => {
+        return '\n';
+    }, []);
```

Comment:
Why aren't this functions part of the removeMarkdown component? I'm 5/5 it should be part of that component, and you could pass optional props to not render emojis and/or breaks

Reply:
I was conflicted about it too but decided against it to keep the component as an abstract. Moving them to the component 👍